### PR TITLE
CORE-8850: Upgrade Kafka Client Libraries

### PR DESCRIPTION
**Pull request by jujoramos**
Bump Kafka Client libraries to the latest available version, 3.3.1_1.

### CODE REVIEW

**Comment by jujoramos** [comment]
Aside from the regular tests executed within the pipeline, I've manually run the `resiliency-tests` with the changes within this `PR` without any new failures reported.

**Review by yarivus — DISMISSED** [review]
LGTM

File: libs/messaging/kafka-message-bus-impl/src/test/kotlin/net/corda/messagebus/kafka/consumer/CordaKafkaConsumerImplTest.kt [hunk 1]
```diff
@@ -271,14 +271,6 @@ class CordaKafkaConsumerImplTest {
         assertThat(positionAfterReset).isEqualTo(0L)
     }
 
-
-    @Test
-    fun testGetPartitionsNullPointerException() {
-        assertThatExceptionOfType(CordaMessageAPIIntermittentException::class.java).isThrownBy {
```
Comment (LWogan) [inline libs/messaging/kafka-message-bus-impl/src/test/kotlin/net/corda/messagebus/kafka/consumer/CordaKafkaConsumerImplTest.kt:277]: We have logic in our CordaKafkaConsumerImpl that throws a CordaMessageAPIIntermittentException when the kafka client consumer.partitionsFor() returns null. If we're saying that this function will never return null then  we should remove the logic in CordaKafkaConsumerImpl to do that. 

Its probably safer to just leave that logic and this test in place though?

Comment (jujoramos) [reply libs/messaging/kafka-message-bus-impl/src/test/kotlin/net/corda/messagebus/kafka/consumer/CordaKafkaConsumerImplTest.kt:277]: Missed that one, will update the `PR` with the changes soon, thanks.

**Review by yarivus — APPROVED** [review]
LGTM

**Review by LWogan — APPROVED** [review]

**Pull request merged**